### PR TITLE
Engine seelog

### PR DESCRIPTION
**Pull request by aaithal**
### Summary

Removed module logger and switched to `seelog` in `engine` package

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: yes

### Description for the changelog

None

### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

### CODE REVIEW

File: agent/engine/docker_container_engine.go [hunk 1]
```diff
@@ -658,7 +665,7 @@ func (dg *dockerGoClient) stopContainer(ctx context.Context, dockerID string) Do
 	err = client.StopContainerWithContext(dockerID, uint(dg.config.DockerStopTimeout/time.Second), ctx)
 	metadata := dg.containerMetadata(dockerID)
 	if err != nil {
-		log.Debug("Error stopping container", "err", err, "id", dockerID)
+		seelog.Infof("Error stopping container %s: %v", dockerID, err)
```
Comment (adnxn) [inline agent/engine/docker_container_engine.go]: `DockerGoClient:` tag for this line

File: agent/engine/docker_task_engine.go [hunk 2]
```diff
@@ -221,7 +221,7 @@ func (engine *DockerTaskEngine) MustInit(ctx context.Context) {
 		err := engine.Init(ctx)
 		if err != nil {
 			errorOnce.Do(func() {
-				log.Error("Could not connect to docker daemon", "err", err)
+				seelog.Errorf("Task engine: could not connect to docker daemon: %v", err)
```
Comment (adnxn) [inline agent/engine/docker_task_engine.go]: nit: should we tag these as `DockerTaskEngine` to keep parity with naming style as we have for `DockerGoClient`?

File: agent/engine/docker_task_engine.go [hunk 3]
```diff
@@ -863,11 +906,13 @@ func (engine *DockerTaskEngine) buildCNIConfigFromTaskContainer(task *api.Task,
 }
 
 func (engine *DockerTaskEngine) stopContainer(task *api.Task, container *api.Container) DockerContainerMetadata {
-	seelog.Infof("Stopping container, container: %s, task: %s", container.String(), task.String())
+	seelog.Infof("Task engine[ %s]: stopping container [%s]", task.Arn, container.Name)
```
Comment (adnxn) [inline agent/engine/docker_task_engine.go]: spacing: `Task engine[ %s]`

File: agent/engine/docker_task_engine.go [hunk 4]
```diff
@@ -882,16 +927,17 @@ func (engine *DockerTaskEngine) stopContainer(task *api.Task, container *api.Con
 	if container.Type == api.ContainerCNIPause {
 		err := engine.cleanupPauseContainerNetwork(task, container)
 		if err != nil {
-			seelog.Errorf("Engine: cleanup pause container network namespace error, task: %s", task.String())
+			seelog.Errorf("Task engine[%s]: unable to cleanup pause container network namespace: %v",
```
Comment (adnxn) [inline agent/engine/docker_task_engine.go]: spacing: `Task engine[%s]`

File: agent/engine/docker_task_engine.go [hunk 5]
```diff
@@ -882,16 +927,17 @@ func (engine *DockerTaskEngine) stopContainer(task *api.Task, container *api.Con
...
+				task.Arn, err)
 		}
-		seelog.Infof("Cleaned pause container network namespace, task: %s", task.String())
+		seelog.Infof("Task engine[%s]: cleaned pause container network namespace", task.Arn)
```
Comment (adnxn) [inline agent/engine/docker_task_engine.go]: spacing: `Task engine[%s]`

File: agent/engine/task_manager.go [hunk 6]
```diff
@@ -165,17 +168,19 @@ func (mtask *managedTask) overseeTask() {
 		if !mtask.GetKnownStatus().Terminal() {
 			// If we aren't terminal and we aren't steady state, we should be
 			// able to move some containers along.
-			llog.Debug("Task not steady state or terminal; progressing it")
+			seelog.Debugf("Managed task [%s]: task not steady state or terminal; progressing it",
```
Comment (adnxn) [inline agent/engine/task_manager.go]: nit: again to keep parity, should we make the tags `ManagedTask`?

File: agent/engine/task_manager.go [hunk 7]
```diff
@@ -358,16 +366,17 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 		now := mtask.time().Now()
 		ok := mtask.Task.SetExecutionStoppedAt(now)
 		if ok {
-			seelog.Infof("Recording execution stopped time for a task, essential container in task stopped, task %s, time: %s",
-				mtask.Task.String(), now.String())
+			seelog.Infof("Managed task [%s]: recording execution stopped time. Essential container %s stopped at: %s",
```
Comment (adnxn) [inline agent/engine/task_manager.go]: nit: `container [%s]`, I think there are few of these where formatting varies from line to line. Not sure if this was intentional.

File: agent/engine/docker_task_engine.go (hunk 2 above)
Comment (aaithal) [reply agent/engine/docker_task_engine.go]: TL;DR I don't think so. Let me know what you think.

I used `DockerGoClient` as I couldn't think of a better name for this. `"Task engine"` distinguishes this from things like `"Stats engine"`. I don't know if we need very strict rules about this.

Comment (adnxn) [reply agent/engine/docker_task_engine.go]: Okay that's fine with me. I don't feel strongly either way. I'm guessing we'll have a better idea after a couple of weeks of log diving and debugging with these new changes 😛

File: agent/engine/dependencygraph/graph.go [hunk 8]
```diff
@@ -23,15 +23,15 @@ import (
...
-	// UnableTransitionContainerPassedDesiredStatus is the error where the container status is bigger than desired status
-	UnableTransitionContainerPassedDesiredStatus = errors.New("container transition: container status is equal or greater than desired status")
-	unableTransitionVolumesDependencyNotResolved = errors.New("dependency graph: container volume dependency not resolved")
-	unableTranstionLinksDependencyNotResolved    = errors.New("dependency graph: container links dependency not resolved")
+	CredentialsNotResolvedErr = errors.New("dependency graph: container execution credentials not available")
+	// DependentContainerNotResolvedErr is the error where a dependent container isn't in expected state
+	DependentContainerNotResolvedErr = errors.New("dependency graph: dependent container not in expected state")
+	// ContainerPastDesiredStatusErr is the error where the container status is bigger than desired status
+	ContainerPastDesiredStatusErr = errors.New("container transition: container status is equal or greater than desired status")
+	volumeUnresolvedErr           = errors.New("dependency graph: container volume dependency not resolved")
```
Comment (sharanyad) [inline agent/engine/dependencygraph/graph.go]: nit: missing comments

File: agent/engine/docker_task_engine.go [hunk 9]
```diff
@@ -336,7 +339,7 @@ func (engine *DockerTaskEngine) synchronizeContainerStatus(container *api.Docker
 func (engine *DockerTaskEngine) CheckTaskState(task *api.Task) {
 	taskContainers, ok := engine.state.ContainerMapByArn(task.Arn)
 	if !ok {
-		seelog.Warnf("Could not check task state for task; no task in state: %s", task.Arn)
+		seelog.Warnf("Task engine [%s]: could not check task state for task; no task in state", task.Arn)
```
Comment (sharanyad) [inline agent/engine/docker_task_engine.go]: rewording suggestion: `could not find task in state`

File: agent/engine/task_manager.go [hunk 10]
```diff
@@ -382,37 +391,43 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
...
-		seelog.Infof("Unable to create container state change event for container '%s' in task '%s': %v",
-			cont.Name, task.Arn, err)
+		seelog.Infof("Managed task [%s]: unable to create state change event for container [%s]: %v",
+			task.Arn, cont.Name, err)
 		return
 	}
 
-	seelog.Infof("Managed task: sending container change event: %s", event.String())
+	seelog.Infof("Managed task [%s]: sending container change event [%s]: %s",
+		mtask.Arn, container.Name, event.String())
```
Comment (sharanyad) [inline agent/engine/task_manager.go]: Seems this should be `cont.Name`

**Pull request merged**